lua/dap/utils.lua

```diff
@@ -0,0 +1,68 @@
+local M = {}
+
+
+local function iscallable(x)
+  if type(x) == 'function' then
+    return true
+  elseif type(x) == 'table' then
+    local mt = debug.getmetatable(x)
+    return type(mt) == "table" and type(mt.__call) == "function"
+  else
+    return false
+  end
+end
+
+
+function M.map_to_key(fn, tbl)
```

Comment:
```suggestion
function M.group_values(values, get_key)
```

Took me a bit to get what the function does. To me `map` feels a bit overloaded in this context because it is often used to refer to the type `Map` (a dictionary, mapping key to values), but also in the context of "mapping a function over values"

Reply:
renamed to `calc_kv_table_from_values`